modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/span/command/EventContextEndSpanCommand.java

```diff
@@ -11,55 +11,38 @@
 
 import org.mule.runtime.api.event.EventContext;
 import org.mule.runtime.tracer.api.context.SpanContext;
-import org.mule.runtime.tracer.api.span.InternalSpan;
 import org.mule.runtime.tracer.api.span.validation.Assertion;
 
-import java.util.Optional;
+import java.util.function.BiConsumer;
+
+import org.slf4j.Logger;
 
 /**
- * A {@link VoidCommand} that ends the current {@link org.mule.runtime.tracer.api.span.InternalSpan}. The carrier is the
- * {@link org.mule.runtime.api.event.EventContext}
+ * A {@link AbstractFailSafeVoidBiCommand} that ends the current {@link org.mule.runtime.tracer.api.span.InternalSpan}. The
+ * carrier is the {@link org.mule.runtime.api.event.EventContext}
  *
  * @since 4.5.0
  */
-public class EventContextEndSpanCommand extends AbstractFailsafeSpanVoidCommand {
-
-  public static final String ERROR_MESSAGE = "Error ending a span";
-  public static final String THREAD_END_SPAN = "thread.end.name";
+public class EventContextEndSpanCommand extends AbstractFailSafeVoidBiCommand<EventContext, Assertion> {
 
-  private final EventContext eventContext;
-  private final Assertion assertion;
-
-  /**
-   * Return a {@link VoidCommand} that ends the current span in the {@link EventContext}
-   *
-   * @param eventContext the {@link EventContext}.
-   * @param assertion    the {@link Assertion} to validate when ending the {@link EventContext}.
-   *
-   * @return the {@link EventContextEndSpanCommand}.
-   */
-  public static EventContextEndSpanCommand getEventContextEndSpanCommandFrom(EventContext eventContext, Assertion assertion) {
-    return new EventContextEndSpanCommand(eventContext, assertion);
+  public static EventContextEndSpanCommand getEventContextEndSpanCommandFrom(Logger logger,
+                                                                             String errorMessage,
+                                                                             boolean propagateException) {
+    return new EventContextEndSpanCommand(logger, errorMessage, propagateException);
   }
 
-  private EventContextEndSpanCommand(EventContext eventContext, Assertion assertion) {
-    this.eventContext = eventContext;
-    this.assertion = assertion;
+  private EventContextEndSpanCommand(Logger logger, String errorMessage, boolean propagateExceptions) {
+    super(logger, errorMessage, propagateExceptions);
   }
 
-  protected Runnable getRunnable() {
-    return () -> {
+  @Override
+  BiConsumer<EventContext, Assertion> getConsumer() {
```

Comment:
To understand a little bit more, on the "getFunction" classes such as AbstractFailsafeUnaryCommand, the command is created on the constructor, saved on a variable and then returned on the getter. Why are we not doing the same for the "getConsumer" classes such as this one? Why is the consumer created on the getter and not on the constructor?

Reply:
Yes, you're right. My mistake. It should be created in the constructor.